contentgrid-appserver-query-engine-impl-jooq/src/main/java/com/contentgrid/appserver/query/engine/jooq/JOOQQueryEngine.java

```diff
@@ -503,10 +526,13 @@ public Optional<EntityIdAndVersion> findTarget(@NonNull Application application,
 
     @Override
     public void setLink(@NonNull Application application, @NonNull RelationRequest relationRequest, @NonNull EntityId targetId,
-            @NonNull ThunkExpression<Boolean> permitUpdatePredicate) throws QueryEngineException {
+            @NonNull ThunkExpression<Boolean> permitUpdatePredicate, @NonNull LinkEventConsumer linkEventConsumer) throws QueryEngineException {
+
+        // Permission check
+        var oldEntityData = getByIdRequired(application, relationRequest, permitUpdatePredicate);
 
-        // implicit permission check + version check
-        var expectedId = findTarget(application, relationRequest, permitUpdatePredicate)
+        // Version check
```

Comment:
This is incorrect; the version check is still being done in `getByIdRequired` (as part of the `findById` method).
This is just retrieving the `ExpectedId`for the relation update itself (not a version check)